tests/unit/job/test_query.py

```diff
@@ -1682,6 +1682,75 @@ def test_result_with_start_index(self):
             tabledata_list_request[1]["query_params"]["maxResults"], page_size
         )
 
+    def test_result_with_start_index_multi_page(self):
+        # When there are multiple pages of response and the user has set
+        # start_index, we should supply start_index to the server in the first
+        # request. However, in the subsequent requests, we will pass only
+        # page_token but not start_index, because the server only allows one
```

Comment:
Here we mention sending `page_token`, but nowhere in the test do we confirm that `page_token` is set or that it has been sent, etc.

In fact, while it is correct, with this language in place, it seems weird that in `tabledata_resource_2`, `pagetoken` is set to `None`. The above language implies that we might send something besides `None`. 

Should we revise the language slightly?
Do we mean some other argument/attribute besides `page_token`?

Reply:
In the last message, the page token will be empty because there is no following page.

Reply:
I am gonna drop this comment.